src/internal/errors/errors.go

```diff
@@ -33,6 +33,20 @@ var (
 	WithStack = errors.WithStack
 )
 
+// UserError is a type used to indicate that an error is not an internal error
+// so that we can log it differently (i.e. do not log at the ERROR level with a
+// stack trace)
+type UserError struct{}
+
+func (err UserError) IsUserError() bool {
+	return true
+}
+
+// Interface for type-checking user errors
+type UserErrorer interface {
```

Comment:
I really dislike this, but couldn't find another way to do this without making it more cumbersome on user code.

Reply:
If you are just relying on the embedding to do a type switch.  You could probably make the interface method private `isUserError()` and make it return nothing.  `UserError` is in the same package, so it can implement it.  Otherwise, when would someone ever implement IsUserError and have it return false? Wouldn't they just not implement it?